### PR TITLE
Bufix/#86 second line

### LINKED ISSUE #86: Bugfix: Second Line Issues
Second line on demo page ('Milano Cortina' text) using ITC Avant Garde missing index on 'M' and 'A' on demo page, accidentally adding ligature to 'I'

### LINKED ISSUE #100: Fix formatting on clear buttons

### PR DESCRIPTION
**Pull request by mattcowan**
Closes #86 , closes #100 

This pull request introduces robust handling of `<br>` (line break) elements in text offset calculations for the Typography Stylist block, ensuring that styling and selections match WordPress RichText's offset system. The main focus is on accurately mapping text positions and applying styles when content includes hard line breaks, fixing previous bugs where styling could be misapplied after `<br>` tags. The update includes new utility functions, test coverage, and integration into the block's editing logic.

**Key changes:**

### Core logic improvements

* Added a new `buildTextOffsetMap` utility function in `utils.js` to generate a mapping of text node positions that treats each `<br>` as a single character offset, matching WordPress RichText's behavior. This ensures that styling and selection offsets are accurate even with line breaks.
* Introduced `getEffectiveTextLength` in `utils.js`, which recursively counts text length in a DOM node, treating `<br>` as one character, for consistent offset calculations.

### Integration into block editor

* Updated `block-editor.js` to use `buildTextOffsetMap` for all offset calculations involving styled spans and text selections, replacing previous logic that did not account for `<br>` elements. This change ensures selections and style applications are correctly aligned with user expectations. [[1]](diffhunk://#diff-9199889613c253a3665b9cdfcfcbb16a2ac54bfb04c6e89cc658dbe9740dfe79R96-R150) [[2]](diffhunk://#diff-9199889613c253a3665b9cdfcfcbb16a2ac54bfb04c6e89cc658dbe9740dfe79R387-L354) [[3]](diffhunk://#diff-9199889613c253a3665b9cdfcfcbb16a2ac54bfb04c6e89cc658dbe9740dfe79L1039-L1060)
* Modified `edit.js` to import and use `buildTextOffsetMap` in the `getRangeForOffsets` helper, ensuring all range calculations in the editor panel handle `<br>` offsets properly. [[1]](diffhunk://#diff-0e056b3f6675fe5e5e405a7b92ff7761acf9fcfe45a7373e588cdaed0682b29cL32-R32) [[2]](diffhunk://#diff-0e056b3f6675fe5e5e405a7b92ff7761acf9fcfe45a7373e588cdaed0682b29cR268-L291)

### Testing and validation

* Added a comprehensive test suite in `__tests__/br-offset.test.js` to verify correct handling of `<br>` elements in offset mapping, text length calculation, feature/style parsing at the cursor, and styling application. These tests cover edge cases such as consecutive `<br>`s, styled spans, and selections spanning line breaks.

### UI adjustments

* Removed unnecessary `float: right` styling from reset/clear buttons in the editor's sidebar for improved layout consistency. [[1]](diffhunk://#diff-0e056b3f6675fe5e5e405a7b92ff7761acf9fcfe45a7373e588cdaed0682b29cL2296-R2291) [[2]](diffhunk://#diff-0e056b3f6675fe5e5e405a7b92ff7761acf9fcfe45a7373e588cdaed0682b29cL2333-R2328) [[3]](diffhunk://#diff-0e056b3f6675fe5e5e405a7b92ff7761acf9fcfe45a7373e588cdaed0682b29cL2398-R2393) [[4]](diffhunk://#diff-0e056b3f6675fe5e5e405a7b92ff7761acf9fcfe45a7373e588cdaed0682b29cL2434-R2429) [[5]](diffhunk://#diff-0e056b3f6675fe5e5e405a7b92ff7761acf9fcfe45a7373e588cdaed0682b29cL2465-R2460)

### Miscellaneous

* Updated `.npmignore` to exclude the `todo/` directory.

These changes collectively resolve issues with misaligned styling and selection when editing text with line breaks and provide a solid foundation for further rich text improvements.

### CODE REVIEW

File: assets/js/block-editor.js [hunk 1]
```diff
@@ -93,6 +93,61 @@ const RESPONSIVE_FONT_MAX_VIEWPORT = 1920; // Desktop baseline
...
+                    end: currentOffset + text.length,
+                    text: text
+                });
+                currentOffset += text.length;
+            } else if (node.nodeName === 'BR') {
+                currentOffset += 1;
+            }
+        }
+
+        return map;
```
[the remark this answers is not in the document]
Comment (mattcowan) [reply assets/js/block-editor.js:148]: Rejecting, because: 

- Introduces a silent failure path. If the load order is wrong, the function returns [], every offset calculation silently breaks, and styles get applied to wrong characters — the exact bug we just fixed. At least with duplication both copies always work.

- Adds a console.warn which violates the project's coding standards (CLAUDE.md explicitly prohibits console logging in production code).

- Creates a runtime dependency on load order between two independently-built bundles that WordPress doesn't guarantee will execute in a specific sequence.
